Switch hardened derivation marker to h (#26076)
bitcoin/bitcoin: doc/release-notes-26076.md

RPC
---

- The `listdescriptors` RPC now uses `h` rather than apostrophe (`'`) to indicate
  hardened derivation. This does not apply when using the `private` parameter, which
  matches the marker used when descriptor was generated or imported. Newly created
  wallets use `h`, but previously created descriptor wallets still use `'`.
  This change makes it easier to handle descriptor strings manually. E.g. an RPC call that

>   takes an array of descriptors can now use '["desc": ".../0h/..."]'. Either `h`

Author: I modified the text a bit, and also updated the PR description.

RPC
---

> - The `listdescriptors` RPC now shows `h` rather than apostrophe (`'`) to indicate

Reviewer: `decodepsbt` too.

RPC
---

- The `listdescriptors`, `decodepsbt` and similar RPC methods now show `h` rather than apostrophe (`'`) to indicate
  hardened derivation. This does not apply when using the `private` parameter, which
  matches the marker used when descriptor was generated or imported. Newly created
  wallets use `h`. This change makes it easier to handle descriptor strings manually.
  E.g. the `importdescriptors` RPC call is easiest to use `h` as the marker: `'["desc": ".../0h/..."]'`.
  With this change `listdescriptors` will use `h`, so you can copy-paste the result,
  without having to add escape characters or switch `'` to 'h' manually.
  Note that this changes the descriptor checksum.

>   For legacy wallets the `hdkeypath` field in `getaddressinfo` is unchanged,
>   nor is the serialization format of wallet dumps. (#26076)

Reviewer 2: The “nor” here seems misplaced:

```suggestion
  For legacy wallets the `hdkeypath` field in `getaddressinfo` 
  and the serialization format of wallet dumps remain unchanged. (#26076)
```

RPC
---

- The `deriveaddresses`, `getdescriptorinfo` and `listdescriptors` RPC's now
  use `h` rather than apostrophe (`'`) to indicate hardened derivation. This

>   makes it easier to handle descriptor strings manually. E.g. an RPC call that

Reviewer 3: The enclosing single quotes are shell escaping, not part of the actual RPC call (which has no issue at present).

Author: By "manually" I mostly mean using `bitcoin-cli` from the shell.
